src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesFactoryAdapter.java

```diff
@@ -132,31 +96,19 @@ public KubernetesClient createClient() throws NoSuchAlgorithmException, Unrecove
             }
         }
 
-        if (credentials instanceof FileCredentials) {
-            // already handled above
-        } else if (credentials instanceof StringCredentials) {
-            final String token = ((StringCredentials) credentials).getSecret().getPlainText();
-            builder.withOauthToken(token);
-        } else if (credentials instanceof TokenProducer) {
-            final String token = ((TokenProducer) credentials).getToken(serviceAddress, caCertData, skipTlsVerify);
-            builder.withOauthToken(token);
-        } else if (credentials instanceof UsernamePasswordCredentials) {
-            UsernamePasswordCredentials usernamePassword = (UsernamePasswordCredentials) credentials;
-            builder.withUsername(usernamePassword.getUsername())
-                    .withPassword(Secret.toString(usernamePassword.getPassword()));
-        } else if (credentials instanceof StandardCertificateCredentials) {
-            StandardCertificateCredentials certificateCredentials = (StandardCertificateCredentials) credentials;
-            KeyStore keyStore = certificateCredentials.getKeyStore();
-            String alias = keyStore.aliases().nextElement();
-            X509Certificate certificate = (X509Certificate) keyStore.getCertificate(alias);
-            Key key = keyStore.getKey(alias, Secret.toString(certificateCredentials.getPassword()).toCharArray());
-            builder.withClientCertData(Base64.encodeBase64String(certificate.getEncoded()))
-                    .withClientKeyData(pemEncodeKey(key))
-                    .withClientKeyPassphrase(Secret.toString(certificateCredentials.getPassword()));
-        } else if (credentials instanceof DockerServerCredentials) {
-            DockerServerCredentials certificateCredentials = (DockerServerCredentials) credentials;
-            builder.withClientCertData(certificateCredentials.getClientCertificate())
-                    .withClientKeyData(certificateCredentials.getClientKey());
+        if (auth instanceof KubernetesAuthToken) {
```

Comment:
Could be implemented directly in `KubernetesAuth` implementations as a method such as `public void decorate(ConfigBuilder builder)`. It would require pushing the kubernetes-client dependency up to `kubernetes-credentials`, or at least the `kubernetes-model` part of it. It has been mentioned in another thread (https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/pull/31) that it would be valuable to extract it into its own library plugin later anyway.

Reply:
Here is similar problem as for kubectl arguments generator method.  For KubernetesAuthKubeconfig we use `builder = new ConfigBuilder(Config.fromKubeconfig`, for other types it is `new ConfigBuilder(Config.autoConfigure`

Reply:
I was given the example of Gitea plugin, so I found there are also `if`s for different types (https://github.com/jenkinsci/gitea-plugin/blob/master/src/main/java/org/jenkinsci/plugin/gitea/client/impl/DefaultGiteaConnection.java#L855)
docker-commons-plugin also uses that way, however, there is only one auth type there

Reply:
Ok